File: biiling-service/src/main/java/com/billing/biiling_service/entities/BillDetail.java
package com.billing.biiling_service.entities;

import com.billing.biiling_service.datas.Client;
import com.billing.biiling_service.datas.Product;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

@Entity
@Table(name = "bill_details")
@Data
@NoArgsConstructor
@AllArgsConstructor

public class BillDetail {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private Long product_id;
    @Transient
    private Product product;
    private Integer quantity;
    private double price;
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private Long bill_id;
}

File: mobile_app/lib/entities/bill_detail.dart
import 'package:billing_app/entities/product.dart';

class BillDetail {
  final int? id;
  final int billId;
  final int productId;
  final int quantity;
  final double price;
  Product? product;

  BillDetail({
    this.id,
    this.billId = 0,
    this.productId = 0,
    required this.quantity,
    required this.price,
    this.product,
  });

  factory BillDetail.fromJson(Map<String, dynamic> json) {
    return BillDetail(
      id: json['id'],
      price: double.parse(json['price'].toString()),
      product:
          json['product'] == null ? null : Product.fromJson(json['product']),
      quantity: int.parse(json['quantity'].toString()),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'product_id': productId,
      'bill_id': billId,
      'quantity': quantity,
      'price': price,
    };
  }
}
